### ISSUE TITLE
Device Posture documentation incorrect

### ISSUE THREAD

**Issue by jamurtag** [cloudflare/terraform-provider-cloudflare#1858]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.2.7
Cloudflare 3.22.0.

### Affected resource(s)

cloudflare_device_posture_integration
cloudflare_device_posture_rule

### Terraform configuration files

```hcl
resource "cloudflare_device_posture_rule" "gateway" {
  account_id  = local.cloudflare_account_id
  name        = "Gateway"
  type        = "gateway"
}
```

### Debug output

n/a

### Expected output

There is a minor documentation error in cloudflare_device_posture_integration - `type` under argument reference only lists `workspace_only`, however other providers are referenced below.

There are more discrepencies in cloudflare_device_posture_rule:
- `type` fields as documented do not match the [API](https://api.cloudflare.com/#device-posture-rules-create-device-posture-rule)
  - tanium, gateway & warp are all missing
- `input` is marked as a required field but this is not the case for gateway at least (and probably warp)

In addition, as I was trying to track down where to configure the gateway device posture check, the Zero Trust Access Admin Logs event name for this API call is `Create device posture integration`, which led to a false trail.

### Actual output

The terraform apply using the code above worked successfully, contradicting the documentation.

### Steps to reproduce

See above.

**Comment by jamurtag** [comment]
In addition, I have created a Intune device posture integration, I am now trying to create a `Service provider checks` resource in Terraform. There is no mention of how to do this in the provider, looking at the API I can see it under [devices/posture](https://api.cloudflare.com/#device-posture-rules-list-device-posture-rules):

```json
$ curl  -k -X GET "https://api.cloudflare.com/client/v4/accounts/<account_id>/devices/posture/<intune_integration_id>" <creds>
{"result":{"hidden":true,"input":{"compliance_status":"compliant","connection_id":"<redacted>"},"match":[],"schedule":"5m","id":"<redacted>","type":"intune","description":"","name":"isCompliant","expiration":null},"success":true
```

I tried to create this in Terraform:

```yaml
resource "cloudflare_device_posture_rule" "intune_is_compliant" {
  account_id  = local.cloudflare_account_id
  name        = "isCompliant"
  type        = "intune"
  input {
    compliance_status = "compliant"
    connection_id     = cloudflare_device_posture_integration.relx_intune.id
  }
  match {}
  schedule    = "5m"
}
```

The resulting output:

```
$ terraform plan
╷
│ Error: expected type to be one of [serial_number file application gateway warp domain_joined os_version disk_encryption firewall workspace_one], got intune
│
│   with cloudflare_device_posture_rule.intune_is_compliant,
│   on cloudflare_device_posture.tf line 18, in resource "cloudflare_device_posture_rule" "intune_is_compliant":
│   18:   type        = "intune"
│
```

The `intune` type is not listed in the API docs or supported in the Terraform provider.

**Comment by jacobbednarz** [comment]
the documentation was converted to automatic generation in #1842 and was released in ([v3.25.0](https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/CHANGELOG.md#3250-october-5th-2022) addressing this one.
